File: invoice-service/src/main/java/com/sudo248/invoiceservice/repository/entity/Invoice.java
package com.sudo248.invoiceservice.repository.entity;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

import javax.persistence.*;

@Getter
@Setter
@ToString
@Entity
@Table(name = "invoices")
@NoArgsConstructor
@AllArgsConstructor
public class Invoice {
    @Id
    @Column(name = "invoice_id")
    private String invoiceId;

    @Column(name = "promotion_id")
    private String promotionId;

    @Column(name = "payment_id")
    private String paymentId;

    @Column(name = "cart_id")
    private String cartId;

    @Column(name = "user_id")
    private String userId;

    @Column(name = "totalPrice")
    private Double totalPrice;

    @Column(name = "status")
    private OrderStatus status;

    @Column(name = "totalPromotionPrice")
    private Double totalPromotionPrice;

    @Column(name = "finalPrice")
    private Double finalPrice;
}

File: client/Soc/app/src/main/java/com/sudo248/soc/domain/entity/invoice/Invoice.kt
package com.sudo248.soc.domain.entity.invoice

import com.sudo248.soc.domain.entity.cart.Cart
import com.sudo248.soc.domain.entity.payment.Payment
import com.sudo248.soc.domain.entity.promotion.Promotion
import com.sudo248.soc.domain.entity.user.User

data class Invoice(
    val invoiceId: String,
    val cart: Cart,
    val payment: Payment? = null,
    val promotion: Promotion? = null,
    val user: User,
    val status: OrderStatus,
    val shipment: Shipment,
    val totalPrice: Double,
    val totalPromotionPrice: Double,
    val finalPrice: Double,
)
